src/main/java/de/tum/in/www1/artemis/service/dto/athena/ProgrammingSubmissionDTO.java

```diff
@@ -20,6 +18,6 @@ public record ProgrammingSubmissionDTO(long id, long exerciseId, String reposito
      */
     public static ProgrammingSubmissionDTO of(long exerciseId, @NotNull ProgrammingSubmission submission, String artemisServerUrl) {
         return new ProgrammingSubmissionDTO(submission.getId(), exerciseId,
-                artemisServerUrl + ATHENA_PROGRAMMING_EXERCISE_REPOSITORY_API_PATH + exerciseId + "/submissions/" + submission.getId() + "/repository");
+                artemisServerUrl + "/api/public/athena/programming-exercises/" + exerciseId + "/submissions/" + submission.getId() + "/repository");
```

Comment:
Honestly, I think outside of repositories, using constants does make a lot of sense. (same above)